bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/relay/RelayHandler.java

```diff
@@ -0,0 +1,278 @@
+        return super.processDeviceInfo(deviceInfo);
+    }
+
+    /**
+     * Dynamically configures the channels according to the device mode.
+     * <p>
+     * Two configurations are possible:
+     * 
+     * <ul>
+     * <li>Power Switch Mode (relay stays on indefinitely when switched on)</li>
+     * <li>Impulse Switch Mode (relay stays on for a configured amount of time and then switches off automatically)</li>
+     * </ul>
+     */
+    private void configureChannels() {
+        if (isInImpulseSwitchMode) {
+            configureImpulseSwitchModeChannels();
+        } else {
+            configurePowerSwitchModeChannels();
+        }
+    }
+
+    private void configureImpulseSwitchModeChannels() {
+        List<String> channelsToBePresent = List.of(CHANNEL_IMPULSE_SWITCH, CHANNEL_IMPULSE_LENGTH,
+                CHANNEL_INSTANT_OF_LAST_IMPULSE);
+        List<String> channelsToBeAbsent = List.of(CHANNEL_POWER_SWITCH);
+        configureChannels(channelsToBePresent, channelsToBeAbsent);
+    }
+
+    private void configurePowerSwitchModeChannels() {
+        List<String> channelsToBePresent = List.of(CHANNEL_POWER_SWITCH);
+        List<String> channelsToBeAbsent = List.of(CHANNEL_IMPULSE_SWITCH, CHANNEL_IMPULSE_LENGTH,
+                CHANNEL_INSTANT_OF_LAST_IMPULSE);
+        configureChannels(channelsToBePresent, channelsToBeAbsent);
+    }
+
+    /**
+     * Re-configures the channels of the associated thing, if applicable.
+     * 
+     * @param channelsToBePresent channels to be added, if not present already
+     * @param channelsToBeAbsent channels to be removed, if present
+     */
+    private void configureChannels(List<String> channelsToBePresent, List<String> channelsToBeAbsent) {
+        List<String> channelsToAdd = channelsToBePresent.stream().filter(c -> getThing().getChannel(c) == null)
+                .toList();
+        List<Channel> channelsToRemove = channelsToBeAbsent.stream().map(c -> getThing().getChannel(c))
+                .filter(Objects::nonNull).map(c -> Objects.requireNonNull(c)).toList();
+
+        if (channelsToAdd.isEmpty() && channelsToRemove.isEmpty()) {
+            return;
+        }
+
+        ThingBuilder thingBuilder = editThing();
+        if (!channelsToAdd.isEmpty()) {
+            addChannels(channelsToAdd, thingBuilder);
+        }
+        if (!channelsToRemove.isEmpty()) {
+            thingBuilder.withoutChannels(channelsToRemove);
+        }
+
+        updateThing(thingBuilder.build());
+    }
+
+    private void addChannels(List<String> channelsToAdd, ThingBuilder thingBuilder) {
+        for (String channelToAdd : channelsToAdd) {
+            Channel channel = createChannel(channelToAdd);
+            thingBuilder.withChannel(channel);
+        }
+    }
+
+    private Channel createChannel(String channelId) {
+        ChannelUID channelUID = new ChannelUID(getThing().getUID(), channelId);
+        ChannelTypeUID channelTypeUID = new ChannelTypeUID(BINDING_ID, channelId);
```

Comment:
It might be cosmetical only, but it seems slighty confusing providing channelId for a channel**Type**Id parameter. Will it work for channel `power-switch` where the statically declared channels refer to channel type `system.power`, so maybe there is no channel type `power-switch`?

Reply:
You're right, I forgot that the association is different for the `power-switch` channel. However, I now have another problem: the constructor of `ChannelTypeUID` requires two colon-separated segments, so `system.power` is not accepted. I now pass `system:power`, but I'm not sure this is the correct approach.

Reply:
You can use `SYSTEM_CHANNEL_TYPE_UID_POWER`, see:
https://github.com/openhab/openhab-core/blob/918b4faa3b3254a993774147f798e8bc59580bf6/bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/DefaultSystemChannelTypeProvider.java#L70C40-L70C69